pkg/prometheus/promcfg.go

```diff
@@ -2733,10 +2733,17 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 			}
 
 			if config.Type != nil {
-				configs[i] = append(configs[i], yaml.MapItem{
-					Key:   "type",
-					Value: config.Type,
-				})
+				switch *config.Type {
+				case "NS":
+					configs[i] = cg.WithMinimumVersion("2.49.0").AppendMapItem(configs[i], "type", config.Type)
+				case "SRV", "A", "AAAA", "MX":
+					configs[i] = append(configs[i], yaml.MapItem{
+						Key:   "type",
+						Value: config.Type,
+					})
+				default:
+					level.Warn(cg.logger).Log("msg", fmt.Sprintf("ignoring DNSSDConfig Type not supported by Prometheus: %s", *config.Type))
+				}
```

Comment:
The operator relies on the API validation to ensure that the value respects the possible enums. To keep code simple, I wouldn't be too defensive here.

```suggestion
				if *config.Type == "NS" {
					configs[i] = cg.WithMinimumVersion("2.49.0").AppendMapItem(configs[i], "type", config.Type)
				} else {
					configs[i] = append(configs[i], yaml.MapItem{
						Key:   "type",
						Value: config.Type,
					})
				}
```

Reply:
Got it. Thanks.